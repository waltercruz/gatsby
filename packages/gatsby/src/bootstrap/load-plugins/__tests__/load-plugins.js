const loadPlugins = require(`../index`)
const slash = require(`slash`)

describe(`Load plugins`, () => {
  /**
   * Replace the resolve path and version string.
   * Resolve path will vary depending on platform.
   * Version can be updated (we use external plugin in default config).
   * Both will cause snapshots to differ.
   */
  const replaceFieldsThatCanVary = plugins =>
    plugins.map(plugin => {
      if (plugin.pluginOptions && plugin.pluginOptions.path) {
        plugin.pluginOptions = {
          ...plugin.pluginOptions,
          path: plugin.pluginOptions.path.replace(
            slash(process.cwd()),
            `<PROJECT_ROOT>`
          ),
        }
      }

      return {
        ...plugin,
        id: ``,
        resolve: ``,
        version: `1.0.0`,
      }
    })

  it(`Load plugins for a site`, async () => {
    let plugins = await loadPlugins({ plugins: [] })

    plugins = replaceFieldsThatCanVary(plugins)

    expect(plugins).toMatchSnapshot()
  })

  it(`Loads plugins defined with an object but without an option key`, async () => {
    const config = {
      plugins: [
        {
          resolve: `___TEST___`,
        },
      ],
    }

    let plugins = await loadPlugins(config)

    plugins = replaceFieldsThatCanVary(plugins)

    expect(plugins).toMatchSnapshot()
  })
})
