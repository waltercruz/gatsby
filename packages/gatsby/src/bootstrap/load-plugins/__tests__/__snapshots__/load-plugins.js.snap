// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load plugins Load plugins for a site 1`] = `
Array [
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "dev-404-page",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "load-babel-config",
    "nodeAPIs": Array [
      "onPreBootstrap",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "internal-data-bridge",
    "nodeAPIs": Array [
      "sourceNodes",
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "prod-404",
    "nodeAPIs": Array [
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "webpack-theme-component-shadowing",
    "nodeAPIs": Array [
      "onCreateWebpackConfig",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "default-site-plugin",
    "nodeAPIs": Array [],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/dev-404-page/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/load-babel-config/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/internal-data-bridge/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/prod-404/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
]
`;

exports[`Load plugins Loads plugins defined with an object but without an option key 1`] = `
Array [
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "dev-404-page",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "load-babel-config",
    "nodeAPIs": Array [
      "onPreBootstrap",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "internal-data-bridge",
    "nodeAPIs": Array [
      "sourceNodes",
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "prod-404",
    "nodeAPIs": Array [
      "onCreatePage",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "webpack-theme-component-shadowing",
    "nodeAPIs": Array [
      "onCreateWebpackConfig",
    ],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "TEST",
    "nodeAPIs": Array [],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "default-site-plugin",
    "nodeAPIs": Array [],
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/dev-404-page/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/load-babel-config/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/internal-data-bridge/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/prod-404/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "__TEST__/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "<PROJECT_ROOT>/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
  Object {
    "browserAPIs": Array [],
    "id": "",
    "name": "gatsby-plugin-page-creator",
    "nodeAPIs": Array [
      "createPagesStatefully",
    ],
    "pluginOptions": Object {
      "path": "/src/pages",
      "pathCheck": false,
      "plugins": Array [],
    },
    "resolve": "",
    "ssrAPIs": Array [],
    "version": "1.0.0",
  },
]
`;
